### PR TITLE
Added User Controller Tests

### PR DESCRIPTION
**Pull request by minipaige02**
I created some slightly different edit/update routes for User and added controller tests. Not sure if I covered all the edge cases, so your input would be most appreciated!

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -12,6 +12,8 @@
   <body>
     <ul>
     <%if session[:user_id]%>
+      <% user = User.find_by(id: session[:user_id])%>
+      <li><%= link_to user.username, current_user_path, class: "btn btn-primary" %>
```
Comment (criacuervos) [inline app/views/layouts/application.html.erb]: thanks for adding this!

File: app/views/users/current.html.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<h2>I'm the current user! My username is: <%=@user.username%></h2>
```
Comment (criacuervos) [inline app/views/users/current.html.erb]: this is a super cool feature :)

File: test/controllers/users_controller_test.rb [hunk 3]
```diff
@@ -1,7 +1,163 @@
 require "test_helper"
 
 describe UsersController do
-  # it "does a thing" do
-  #   value(1+1).must_equal 2
-  # end
+  describe "current" do
```
Comment (criacuervos) [inline test/controllers/users_controller_test.rb]: great job, thanks !
